Realm/Tests/DynamicTests.m

```diff
@@ -161,7 +161,7 @@ - (void)testDynamicTypes {
 
     // verify properties
     RLMRealm *dyrealm = [self realmWithTestPathAndSchema:nil];
-    RLMResults *results = [dyrealm allObjects:AllTypesObject.className];
+    RLMResults<RLMObject *> *results = [dyrealm allObjects:AllTypesObject.className];
```

Comment:
Are these changes required by this PR, or are they being made preemptively in advance of #5149?